source/core.py

```diff
@@ -79,7 +79,8 @@ def doStartupDialogs():
 			_("Configuration File Error"),
 			wx.OK | wx.ICON_EXCLAMATION)
 	if config.conf["general"]["showWelcomeDialogAtStartup"]:
-		gui.WelcomeDialog.run()
+		from gui.startupDialogs import WelcomeDialog
+		WelcomeDialog.run()

```

Comment:
```suggestion
		gui.startupDialogs.WelcomeDialog.run()
```

I think we are trying to avoid these imports that aren't at the top of file, or the most outer scope when this isn't possible. Since `gui` has already been imported at the top of `doStartupDialogs`, if you want to import `WelcomeDialog` directly, it would be preferable to add the `from gui.startupDialogs import WelcomeDialog` statement top the top of `doStartupDialogs`.

Reply:
It is a matter of style obviously, however since `WelcomeDialog` is used only for users who haven't disabled it so a very small percentage isn't it wasteful to always import it rather than do it conditionally as it is done now?

Reply:
yes, definitely a matter of preference. From a discussion with another developer these are fine as is. However in this case wouldn't there be no performance changes because the entire `gui` module has already been imported in the same scope?